src/platform/webengine/BlinkWebViewProfileHelper.h

```diff
@@ -40,14 +43,22 @@ const char kWebSQL[] = "webSQL";
 
 class BlinkWebViewProfileHelper {
 public:
-    BlinkWebViewProfileHelper() {}
-    virtual ~BlinkWebViewProfileHelper() {}
+    static BlinkWebViewProfileHelper* instance();
 
     static void clearBrowsingData(const int removeBrowsingDataMask,
         webos::WebViewProfile* profile = nullptr);
     static void clearDefaultBrowsingData(const int removeBrowsingDataMask);
     static int maskForBrowsingDataType(const char* key);
-    static void setProxyRules(const std::string& proxy_rules);
+    webos::WebViewProfile* getProfile(const std::string& app_id);
+    void buildProfile(const std::string& app_id, const std::string& proxy_rules);
+
+private:
+    BlinkWebViewProfileHelper() {}
+    virtual ~BlinkWebViewProfileHelper() = default;
```

Comment:
Now that `BlinkWebViewProfileHelper` is a singleton, there should be an explicit delete of the object. I guess its lifecycle should be tied to that of `WebAppManagerServiceAGL` but I see there is no explicit deletion of the service @jdapena

Reply:
It is ok for me. We have several other singletons handled the same way.